src/types.js

```diff
@@ -9,6 +10,22 @@ const types = {
   strategies: {},
 
   /**
+   * Checks for and applies sub-type to definition
+   * @param {Object} def The rule defintion
+   * @returns {Object}
+   */
+  checkSubType: def => {
+    if (def.type.indexOf(':') >= 0) {
+      const fullType = def.type.split(':')
```

Comment:
You can save yourself an O(n) operation here by putting this line _above_ the if statement, and checking for `fullType.length` in the if-conditional instead of doing the `indexOf`.